src/server/editorServices.ts

```diff
@@ -756,6 +756,11 @@ namespace ts.server {
         /*@internal*/
         readonly watchFactory: WatchFactory<WatchType, Project>;
 
+        /*@internal*/
+        private sharedExtendedConfigFileMap = createMultiMap<Path, ConfiguredProject>();
+        /*@internal*/
+        private sharedExtendedConfigFileWatchers = new Map<Path, FileWatcher>();

```

Comment:
Just make this one map with { watcher: FileWatcher; projects: Set<NormalizedPath>} // Where projects is set of project.canonicalConfigFilePath`